pkg/api/vminfo.go

```diff
@@ -180,49 +182,38 @@ func (vm *VmInfo) applyBounds(b scalingBounds) {
 }
 
 type scalingBounds struct {
-	Min *resourceBound `json:"min,omitempty"`
-	Max *resourceBound `json:"max,omitempty"`
+	Min resourceBound `json:"min"`
+	Max resourceBound `json:"max"`
 }
 
 type resourceBound struct {
-	CPU *uint16            `json:"cpu,omitempty"`
-	Mem *resource.Quantity `json:"mem,omitempty"`
+	CPU uint16            `json:"cpu"`
+	Mem resource.Quantity `json:"mem"`
 }
 
 func (b scalingBounds) validate(memSlotSize *resource.Quantity) error {
-	if b.Min == nil {
-		return errors.New("missing field 'min'")
-	} else if b.Max == nil {
-		return errors.New("missing field 'max'")
-	}
+	ec := erc.Collector{}
 
-	if field, err := b.Min.validate(memSlotSize); err != nil {
-		return fmt.Errorf("error at .min%s: %w", field, err)
-	} else if field, err := b.Max.validate(memSlotSize); err != nil {
-		return fmt.Errorf("error at .max%s: %w", field, err)
-	}
+	b.Min.validate(&ec, ".min", memSlotSize)
+	b.Max.validate(&ec, ".max", memSlotSize)
 
-	return nil
+	return ec.Resolve()
 }
 
-func (b resourceBound) validate(memSlotSize *resource.Quantity) (field string, _ error) {
-	if b.CPU == nil {
-		return "", errors.New("missing field 'cpu'")
-	} else if b.Mem == nil {
-		return "", errors.New("missing field 'mem'")
+func (b resourceBound) validate(ec *erc.Collector, path string, memSlotSize *resource.Quantity) {
+	errAt := func(field string, err error) error {
+		return fmt.Errorf("error at %s%s: %w", path, field, err)
```

Comment:
this does seem like a better solution, but for now, I'd rather take the tech debt as-is than spend time making it better — seems just a bit too complex for me to deal with right now.

Left a note about this so that it's not forgotten. Feel free to handle it post-merge if you particularly want to. I'll probably get around to it _eventually_ if not.